### PR TITLE
Configuration management

### PR DESCRIPTION
**Pull request by dboltovskyi**
This proposal is about providing a set of scripts to easy manage our build/test system.

### CODE REVIEW

**Comment by vkushnirenko-luxoft** [comment]
@dboltovskyi @LuxoftAKutsan @Itileda please review

File: proposals/xxxx-continous-integration.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Configuration management
+
+* Proposal: [SDL-NNNN](xxxx-Configuration-Management.md)
+* Author: [Vadym Kushnirenko](https://github.com/vkushnirenko-luxoft)
+* Status: **Awaiting review**
+* Impacted Platforms: [Build server]
+
+## Introduction
+
+This proposal is about an order of creating build jobs in our buiding system, running tests and keeping results of theese tests.
```
Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Typo 'theese'

File: proposals/xxxx-continous-integration.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+## Motivation
+
+The building process should be transparent. Any person should be able to deploy build server on his side, use it and propose recomendations about improvements.
+Build server configuration must be saved into files, be flexible and be applied in any moment from public resources.
+
+## Proposed solution
+
+Key components of build system are:
+- jenkins master - it manages build/test processes,
+- jenkins slaves - servers which builds and tests,
```
Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft It's better to rephrase 'builds and tests' into something else, e.g. 'servers that allows to build SDL with different options and to run unit and ATF tests'

File: proposals/xxxx-continous-integration.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+- artifactory - keeps archive with binaries and log files.
+- Github - keeps Dockerfiles and Jenkins jobs, allows to keep different versions of configuration and to download a configuration at any time.
+The solution proposes to use docker containers for components deployment. 
+### Jenkins jobs creating.
```
Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Here and in other section names it's better not to put '.' (full stop) at the end

File: proposals/xxxx-continous-integration.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Every task that implies alterations in Git repository should be represented as separated branch. The template for naming the branch is the following:
+feature/<task name> , fix/<task name> or hotfix/<task name>
+All changes related to the task should be applied to this branch.
+The next list of jobs will be created after creating new branch:
+- push,
```
Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Please add some description of jobs

File: proposals/xxxx-continous-integration.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ATF feature test and ATF smoke test sets run after nightly job.
+
+Summary, the next set of jenkins job will be created for each feature/fix/hotfix branch:
+- PUSH_feature
+  - ATF_feature
+- PR_feature
+  - ATF_SMOKE_feature
+- NIGHTLY_feature
+  - ATF_feature
+  - ATF_SMOKE_feature
```
Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft In this list of jobs the one related to SDL build is missed

File: proposals/xxxx-continous-integration.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+Key components of build system are:
+- jenkins master - it manages build/test processes,
+- jenkins slaves - servers which builds and tests,
+- artifactory - keeps archive with binaries and log files.
+- Github - keeps Dockerfiles and Jenkins jobs, allows to keep different versions of configuration and to download a configuration at any time.
+The solution proposes to use docker containers for components deployment. 
+### Jenkins jobs creating.
+Every task that implies alterations in Git repository should be represented as separated branch. The template for naming the branch is the following:
+feature/<task name> , fix/<task name> or hotfix/<task name>
```
Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Are we really expect creating of jobs for fix/hotfixes?

Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Seems ```<task name>``` was removed during formatting

File: proposals/xxxx-continous-integration.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+- PUSH_feature
+  - ATF_feature
+- PR_feature
+  - ATF_SMOKE_feature
+- NIGHTLY_feature
+  - ATF_feature
+  - ATF_SMOKE_feature
+### 1) Jenkins - continuous integration system.
+https://jenkins.io/
+Jenkins is the system for automation routine operations:
```
Comment (dboltovskyi) [inline proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Suggest to rephrase start of the items (here and below) in imperative way, e.g.: 
  - Use Jenkins for ...
  - Use Docker containers ...

File: proposals/xxxx-continous-integration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Accept

[the remark this answers is not in the document]
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Accept

Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Accept

File: proposals/xxxx-continous-integration.md (hunk 2 above)
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Accept

File: proposals/xxxx-continous-integration.md (hunk 3 above)
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Accept

File: proposals/xxxx-continous-integration.md (hunk 6 above)
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Rejected. This is about branch names

Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Rejected. What do you mean?

File: proposals/xxxx-continous-integration.md (hunk 4 above)
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Accepted. Added short description

File: proposals/xxxx-continous-integration.md (hunk 5 above)
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Rejected, please explain.

File: proposals/xxxx-continous-integration.md (hunk 7 above)
Comment (vkushnirenko-luxoft) [reply proposals/xxxx-continous-integration.md]: Rejected, please explain.

File: proposals/xxxx-continous-integration.md (hunk 6 above)
Comment (dboltovskyi) [reply proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Ok

Comment (dboltovskyi) [reply proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft When .md file is in view mode the following is displayed:
" The template for naming the branch is the following: feature/ , fix/ or hotfix/ "
```<task name>``` is removed during formatting

File: proposals/xxxx-continous-integration.md (hunk 5 above)
Comment (dboltovskyi) [reply proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Currently there are only jobs related to ATF
But the job which creates particular SDL build is missing

File: proposals/xxxx-continous-integration.md (hunk 7 above)
Comment (dboltovskyi) [reply proposals/xxxx-continous-integration.md]: @vkushnirenko-luxoft Proposal is about some actions that needs to be done
Please suggest where is an action points for steps 1), 2) etc.
Currently this is just a description of the tools

File: proposals/xxxx-continous-integration.md (hunk 5 above)
Comment (dboltovskyi) [reply proposals/xxxx-continous-integration.md]: It was agreed that currently shown structure of jobs, but not folders

**Comment by dboltovskyi** [comment]
Closed since rework is required

**Pull request closed without merging**